### PR TITLE
9926: list role assignments permissions

### LINKED ISSUE #9926: API vs UI: inconsistency between required permissions for listing role assignments
**What steps does it take to reproduce the issue?**

Via UI:

* Open dataset you have ManageDatasetPermissions permission for
* Click Edit dataset > Permissions > Dataset
* See list of users/groups with assigned roles

Via API:

* GET request for api/datasets/<id>/assignments for dataset with id you have ManageDatasetPermissions permission for
* Receive permissions error

I have searched through the code and I think that is because the ManageDataverse permission is required?

https://github.com/IQSS/dataverse/blob/v6.0/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/ListRoleAssignments.java#L18

**What did you expect to happen?**

I expected the API call to succeed

**Which version of Dataverse are you using?**

5.13

**Any related open or closed issues to this bug report?**

No, I didn't find any

**Screenshots:**

List of role assignments in UI:

image

### PR DESCRIPTION
**Pull request by vera**
**What this PR does / why we need it**:

This fixes the permission inconsistency described in #9926

Listing dataverse role assignments via API still requires ManageDataversePermissions, but listing dataset role assignments via API now requires only the ManageDatasetPermissions.

**Which issue(s) this PR closes**:

Closes #9926

**Special notes for your reviewer**:

/

**Suggestions on how to test this**:

`mvn clean test -Dtest="DatasetsIT#testListRoleAssignments"`

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

no

**Is there a release notes update needed for this change?**:

yes, might be good

**Additional documentation**:

/

### CODE REVIEW

**Comment by johannes-darms** [comment]
@pdurbin : I was wondering if it would be possible to have a review of this MR.

**Comment by pdurbin** [comment]
@scolapasta said he'd take a look.

**Review by scolapasta — CHANGES_REQUESTED** [review]
Looks good, but I left a comment that I hope can remove the need fo rthe contributorPlus role.

**Comment by scolapasta** [comment]
@vera checking in, were you able to loo at my review comments?

**Comment by vera** [comment]
@scolapasta sorry, somehow I missed your comment! Will take a look at this asap

File: src/test/java/edu/harvard/iq/dataverse/api/DatasetsIT.java [hunk 1]
```diff
@@ -1296,6 +1296,93 @@ public void testAddRoles(){
...
+
+        String protocol1 = JsonPath.from(datasetAsJson.getBody().asString()).getString("data.protocol");
+        String authority1 = JsonPath.from(datasetAsJson.getBody().asString()).getString("data.authority");
+        String identifier1 = JsonPath.from(datasetAsJson.getBody().asString()).getString("data.identifier");
+        String datasetPersistentId = protocol1 + ":" + authority1 + "/" + identifier1;
+
+        Response notPermittedToListRoleAssignmentOnDataset = UtilIT.getRoleAssignmentsOnDataset(datasetId.toString(), null, contributorApiToken);
+        assertEquals(UNAUTHORIZED.getStatusCode(), notPermittedToListRoleAssignmentOnDataset.getStatusCode());
+
+        // We create a new role that includes "ManageDatasetPermissions" which are required for listing role assignments
```
[the remark this answers is not in the document]
Comment (vera) [reply src/test/java/edu/harvard/iq/dataverse/api/DatasetsIT.java]: :+1: done!

**Comment by vera** [comment]
@scolapasta I removed the new contributorPlus role and used the curator role instead for the test as you suggested.

**Review by scolapasta — APPROVED** [review]

**Comment by pdurbin** [comment]
@vera hi! Our CI is [failing](https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-9956/3/consoleFull) with this:

`TASK [dataverse : copy warfile over release] ***********************************`
`fatal: [localhost]: FAILED! => {"changed": false, "msg": "Source /opt/dataverse/dataverse/target/dataverse-6.1.war not found"}`

I'll go ahead and merge the latest from develop into your branch to pick up the 6.1 version change.

**Pull request merged**